Tracks/reapertips_Track-Icon-Selector.lua

```diff
@@ -0,0 +1,539 @@
+-- @description Track Icon Selector
+-- @author Reapertips & Sexan
+-- @version 1.0
+-- @screenshot
+--   https://i.imgur.com/bFK2HYk.png
+--   https://i.imgur.com/MabMOW1.png
+-- @about
+--   # Track Icon Selector
+--   Created by Reapertips and Sexan
+--
+--   The Track Icon Selector is a tool designed to streamline the process of applying custom icons to your tracks in REAPER. By scanning REAPER's "track_icons" folder, the script automatically populates a user-friendly interface with icons and their categories.
+--
+--   ## Key Features:
+--
+--   - Intuitive Interface: Quickly browse and select icons from a visually appealing interface.
+--   - Instant Search: Efficiently find specific icons using the built-in search function.
+--   - Categorization: Organize icons into categories for easy navigation.
+--   - Customizable Icons: Resize icons to fit your preferences and customize their appearance.
+--   - Sidebar Preferences: Control whether the sidebar opens automatically on script launch
+--   - Dockable Window: Conveniently dock the script window to the left of your tracks for easy access.
+--   - Yellow Outline Around Used Icons: Instantly identify tracks that already have icons applied.
+--
+--   ## Creating and Modifying Categories:
+--
+--   To create or modify categories for your icons, navigate to the following REAPER folder:
+--
+--   REAPER Folder > Data > track_icons
+--   Within this folder, create or modify folders to represent your desired categories. The script will automatically recognize these folders as categories in the sidebar, allowing you to filter icons based on their assigned categories.
+
+-- @description Track Icon Selector
+-- @author Sexan
+-- @license GPL v3
+-- @version 1.0
+-- @changelog
+--  Initial release
+
+local r                = reaper
+package.path           = r.ImGui_GetBuiltinPath() .. '/?.lua'
+local os_separator     = package.config:sub(1, 1)
+
+local imgui            = require "imgui" "0.9.1"
+local reaper_path      = r.GetResourcePath()
+
+local function NeedDeps()
+    local deps = {}
+
+    if not r.ImGui_GetVersion then
+        deps[#deps + 1] = '"Dear Imgui"'
+    end
+    if #deps ~= 0 then
+        r.ShowMessageBox("Script needs ReaImGui.\nPlease Install it in next window", "MISSING DEPENDENCIES", 0)
+        r.ReaPack_BrowsePackages(table.concat(deps, " OR "))
+        return true
+    end
+end
+if NeedDeps() then return end
```

Comment:
The script will crash prior to reaching here, at the `ImGui_GetBuiltinPath`/ `require` lines above, if ReaImGui is not installed. The check should be done before that, and preferably be looking for `ImGui_GetBuiltinPath` (the only function guaranteed to always exist before enabling backward compat in "0.9.1" mode).

```lua
if not reaper.ImGui_GetBuiltinPath then
  return reaper.MB('ReaImGui is not installed or too old.', 'Track icon selector, 0)
end

package.path = reaper.ImGui_GetBuiltinPath() .. '/?.lua'
local imgui = require 'imgui' '0.9.2'
```